ctapipe/image/ellipsoid.py

```diff
@@ -0,0 +1,494 @@
+            (width_min.to_value(unit), width_max.to_value(unit)),
+            (skew_min, skew_max),
+            (ampl_min, ampl_max),
+        ]
+
+    return [
+        (cogx_min.to_value(unit), cogx_max.to_value(unit)),
+        (cogy_min.to_value(unit), cogy_max.to_value(unit)),
+        (psi_min, psi_max),
+        (length_min.to_value(unit), length_max.to_value(unit)),
+        (width_min.to_value(unit), width_max.to_value(unit)),
+        (ampl_min, ampl_max),
+    ]
+
+
+def boundaries(geometry, image, dilated_mask, x0, pdf):
+    """
+    Computes the boundaries of the fit.
+    Parameters
+    ----------
+    geometry : ctapipe.instrument.CameraGeometry
+        Camera geometry
+    image : ndarray
+        Charge in each pixel, no cleaning mask should be applied
+    dilated_mask : boolean
+        mask after image cleaning and dilation
+    x0 : dict
+       seeds of the fit
+    pdf: str
+       name of the PDF, options = "Gaussian", "Laplace", "Skewed"
+    Returns
+    -------
+    list of boundaries
+    """
+    x = geometry.pix_x.value
+    y = geometry.pix_y.value
+    unit = geometry.pix_x.unit
+    camera_radius = geometry.guess_radius().to_value(unit)
+    leakage = leakage_parameters(geometry, image, dilated_mask)
+
+    # Dilated image
+    row_image = image.copy()
+    row_image[~dilated_mask] = 0.0
+    row_image[row_image < 0] = 0.0
+
+    max_x = np.max(x[dilated_mask])
+    min_x = np.min(x[dilated_mask])
+    max_y = np.max(y[dilated_mask])
+    min_y = np.min(y[dilated_mask])
+
+    psi_min, psi_max = max(x0["psi"].value - 0.2, -np.pi / 2), min(
+        x0["psi"].value + 0.2, np.pi / 2
+    )
+
+    cogx_min, cogx_max = np.sign(min_x) * min(np.abs(min_x), camera_radius), np.sign(
+        max_x
+    ) * min(np.abs(max_x), camera_radius)
+
+    cogy_min, cogy_max = np.sign(min_y) * min(np.abs(min_y), camera_radius), np.sign(
+        max_y
+    ) * min(np.abs(max_y), camera_radius)
+
+    if (leakage.intensity_width_1 > 0.2) & (
+        leakage.intensity_width_2 > 0.2
+    ):  # truncated
+        if (x0["x"] > 0) & (x0["y"] > 0):
+            max_x = 2 * max_x
+            max_y = 2 * max_y
+        if (x0["x"] < 0) & (x0["y"] > 0):
+            min_x = 2 * min_x
+            max_y = 2 * max_y
+        if (x0["x"] < 0) & (x0["y"] < 0):
+            min_x = 2 * min_x
+            min_y = 2 * min_y
+        if (x0["x"] > 0) & (x0["y"] < 0):
+            max_x = 2 * max_x
+            min_y = 2 * min_y
+
+    long_dis = np.sqrt((max_x - min_x) ** 2 + (max_y - min_y) ** 2)
+
+    width_unc = u.Quantity(0.05, unit)
+    length_min, length_max = x0["length"].value, long_dis
+    width_min, width_max = x0["width"].value, x0["width"].value + width_unc.to_value(
+        unit
+    )
+
+    scale = length_min / np.sqrt(1 - 2 / np.pi)
+    skew_min, skew_max = max(-0.99, x0["skewness"] - 0.3), min(
+        0.99, x0["skewness"] + 0.3
+    )
+
+    if pdf == "Gaussian":
```

Comment:
this then becomes, e.g. `PDFType.Gaussian`

Reply:
Is that what you were asking for?